app/controllers/statistics_controller.rb

```diff
@@ -13,11 +13,28 @@ def index
     respond_to do |format|
       format.html
       format.json do
-        render :json => { :statistics => { :os_count    => @os_count, :arch_count => @arch_count,
-                                           :env_count   => @env_count, :klass_count => @klass_count, :cpu_count => @cpu_count,
-                                           :model_count => @model_count, :mem_size => @mem_size, :mem_free => @mem_free, :swap_size => @swap_size,
-                                           :swap_free   => @swap_free, :mem_totsize => @mem_totsize, :mem_totfree => @mem_totfree } }
+        render :json => {:statistics => [
+            buildChartDataJson("os_dist", "OS Distribution", @os_count, "hosts?search=os_title=~VAL~"),
```

Comment:
note, you are doing all of the stats calculation for both html and json responses, it should be moved only to the json part.

also, while the json is loading, it should show either an empty chart, or a loading indicator of some sort.